packages/upload/src/vaadin-upload-file-mixin.js

```diff
@@ -151,7 +161,12 @@ export const UploadFileMixin = (superClass) =>
      * @protected
      */
     _shouldSetFocus(event) {
-      return event.composedPath()[0] === this;
+      return !this.disabled && event.composedPath()[0] === this;
```

Comment:
This only hides the focus outline while the element can still receive focus when triggered programmatically. I checked and this doesn't happen in other components. Should we override the `focus()` method instead?

Reply:
Good point. It actually makes more sense to remove the tabindex attribute when the element is disabled. That makes it non-focusable, but also required changing the property / observer setup a bit. Change is in https://github.com/vaadin/web-components/pull/8090/commits/15acbda73a955fe45ce68ebd560ccaf79f87c77e